### PR TITLE
fix: `UA_Client_MonitoredItems_modify_async` implementation

### PR DESCRIPTION
**Pull request by lukasberbuer**
As discussed in #6269.

### CODE REVIEW

File: src/client/ua_client_subscriptions.c [hunk 1]
```diff
@@ -947,39 +947,64 @@ UA_MonitoredItem_change_clientHandle(void *data, UA_Client_MonitoredItem *mon) {
...
 
     UA_LOCK(&client->clientMutex);
-    UA_Client_Subscription *sub;
-    LIST_FOREACH(sub, &client->subscriptions, listEntry) {
-        if (sub->subscriptionId == request.subscriptionId)
-            break;
-    }
-
+    UA_Client_Subscription *sub = findSubscription(client, request.subscriptionId);
+    UA_UNLOCK(&client->clientMutex);
```
Comment (jpfr) [inline src/client/ua_client_subscriptions.c:968]: Don’t unlock here in the good case.
You continue to use the sub further down.

File: src/client/ua_client_subscriptions.c [hunk 2]
```diff
@@ -947,39 +947,64 @@ UA_MonitoredItem_change_clientHandle(void *data, UA_Client_MonitoredItem *mon) {
...
+    return response;
+}
 
+UA_StatusCode
+UA_Client_MonitoredItems_modify_async(UA_Client *client,
+                                      const UA_ModifyMonitoredItemsRequest request,
+                                      UA_ClientAsyncServiceCallback callback,
+                                      void *userdata, UA_UInt32 *requestId) {
+    UA_LOCK(&client->clientMutex);
+    UA_Client_Subscription *sub = findSubscription(client, request.subscriptionId);
```
Comment (jpfr) [inline src/client/ua_client_subscriptions.c:992]: Also don’t unlock here in the good case.

**Review by jpfr — COMMENTED** [review]
Looks good.
Two comments with respect to locking.

File: src/client/ua_client_subscriptions.c (hunk 1 above)
Comment (lukasberbuer) [reply src/client/ua_client_subscriptions.c]: Fixed in https://github.com/open62541/open62541/pull/6297/commits/0552466662d387814e912e3ce22114befea5b305

File: src/client/ua_client_subscriptions.c (hunk 2 above)
Comment (lukasberbuer) [reply src/client/ua_client_subscriptions.c:993]: Fixed in https://github.com/open62541/open62541/pull/6297/commits/0552466662d387814e912e3ce22114befea5b305

**Comment by jpfr** [comment]
Nice.
As this is a fix and doesn't change the public API, I would like to get this on the 1.4 branch.
From there we forward-merge to master.

However, changing the base for this pull request added a lot of other commits from master.
Please rebase on 1.4. Then we can pull.

**Comment by lukasberbuer** [comment]
👍 Should be ready now.

**Comment by jpfr** [comment]
Great. Thanks for the contribution.

**Pull request merged**